packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -154,15 +277,48 @@ export class DetachedFieldIndex {
 						undefined,
 					0x7ce /* Detached node ID already exists in index */,
 				);
-				setInNestedMap(this.detachedNodeToField, nodeId.major, nodeId.minor + i, root + i);
+				setInNestedMap(this.detachedNodeToField, nodeId.major, nodeId.minor + i, {
+					root: root + i,
+					latestRelevantRevision: revision,
+				});
+				this.updateLatestRevision({ major: nodeId.major, minor: nodeId.minor + i }, revision);
 			}
 		}
 		return root;
 	}
 
+	/**
+	 * Updates the latest revision that is relevant to the provided root
+	 */
```

Comment:
Does this really need to go through the update codepath instead of just setting the data for what should be the first time?

Reply:
Done. Resolving.